src/main/scala/mesosphere/marathon/api/akkahttp/EntityMarshallers.scala

```diff
@@ -95,10 +96,49 @@ object EntityMarshallers {
     jsonStringMarshaller.compose(printer).compose(internal => writes.writes(ramlWrites.write(internal)))
 
   import mesosphere.marathon.raml.AppConversion.appRamlReader
-  implicit def appUnmarshaller(
+  implicit def appDefinitionUnmarshaller(
     implicit
-    normalization: Normalization[raml.App], validator: Validator[AppDefinition]) = {
-    validEntityRaml(playJsonUnmarshaller[raml.App])
+    normalization: Normalization[raml.App],
+    validator: Validator[AppDefinition]): FromEntityUnmarshaller[AppDefinition] = {
+    validEntityRaml(playJsonUnmarshaller[raml.App]).handleValidationErrors
```

Comment:
How about moving these changes to a separated diff as well?

Reply:
Should we review it as is? Breaking this PR into multiple will slow down us, IMHO, since a lot of time has been invested into reviewing this already.